ocs_ci/ocs/registry.py

```diff
@@ -0,0 +1,282 @@
+
+
+logger = logging.getLogger(__name__)
+
+
+def change_registry_backend_to_ocs():
+    """
+    Function to deploy registry with OCS backend.
+
+    Raises:
+        AssertionError: When failure in change of registry backend to OCS
+    """
+    pv_obj = helpers.create_pvc(
+        sc_name=constants.DEFAULT_SC_CEPHFS, pvc_name='registry-cephfs-rwx-pvc',
+        namespace=constants.OPENSHIFT_IMAGE_REGISTRY_NAMESPACE, size='100Gi',
+        access_mode=constants.ACCESS_MODE_RWX
+    )
+    helpers.wait_for_resource_state(pv_obj, 'Bound')
+    ocp_obj = ocp.OCP(
+        kind=constants.CONFIG, namespace=constants.OPENSHIFT_IMAGE_REGISTRY_NAMESPACE
+    )
+    param_cmd = f'[{{"op": "add", "path": "/spec/storage", "value": {{"pvc": {{"claim": "{pv_obj.name}"}}}}}}]'
+    assert ocp_obj.patch(
+        resource_name=constants.IMAGE_REGISTRY_RESOURCE_NAME, params=param_cmd
+    ), f"Registry pod storage backend to OCS is not success"
+
+    # Validate registry pod status
+    validate_registry_pod_status()
+
+    # Validate pvc mount in the registry pod
+    validate_pvc_mount_on_registry_pod()
+
+
+def get_registry_pod_obj():
+    """
+    Function to get registry pod obj
+
+    Returns:
+        pod_obj (obj): Registry pod obj
+
+    Raises:
+        UnexpectedBehaviour: When image-registry pod is not present.
+    """
+    # Sometimes when there is a update in config crd, there will be 2 registry pods
+    # i.e. old pod will be terminated and new pod will be up based on new crd
+    # so below loop waits till old pod terminates
+    wait_time = 30
+    iteration = 0
+    while True:
+        iteration += 1
+        pod_data = pod.get_pods_having_label(
+            label='docker-registry=default', namespace=constants.OPENSHIFT_IMAGE_REGISTRY_NAMESPACE
+        )
+        pod_obj = [pod.Pod(**data) for data in pod_data]
+        if len(pod_obj) == 1:
+            break
+        elif len(pod_obj) == 0:
+            raise UnexpectedBehaviour("Image-registry pod not present")
+        elif iteration > 6:
+            raise UnexpectedBehaviour("Waited for 3 mins Image-registry pod is not in Running state")
+        else:
+            logger.info(f"Waiting for 30 sec's for registry pod to be up iteration {iteration}")
+            time.sleep(wait_time)
+    return pod_obj
+
+
+def get_oc_podman_login_cmd():
+    """
+    Function to get oc and podman login commands on node
+
+    Returns:
+        cmd_list (list): List of cmd for oc/podman login
+    """
+    user = config.RUN['username']
+    filename = os.path.join(
+        config.ENV_DATA['cluster_path'],
+        config.RUN['password_location']
+    )
+    with open(filename) as f:
+        password = f.read()
+    helpers.refresh_oc_login_connection()
+    ocp_obj = ocp.OCP()
+    token = ocp_obj.get_user_token()
+    route = get_default_route_name()
+    cmd_list = [
+        'export KUBECONFIG=/home/core/auth/kubeconfig',
+        f"oc login -u {user} -p {password}",
```

Comment:
If we have kubeconfig exported why the login is needed?

Reply:
This set of commands will be executed inside the NODE. In debug mode kubeconfig export, oc login is required for each podman command execution, else podman commands will fail.